go-controller/pkg/node/helper_linux.go

```diff
@@ -36,7 +36,12 @@ func getDefaultGatewayInterfaceDetails(gwIface string) (string, []net.IP, error)
 		if err != nil {
 			return "", gatewayIPs, err
 		}
+
 		// validate that both IP Families use the same interface for the gateway
+		// if we found a v4 default gw inf then we should expect v6 default gw inf also
+		if intfName != "" && intfIPv6Name == "" {
+			return "", nil, fmt.Errorf("failed to find IPV6 default gateway from interface %q", gwIface)
```

Comment:
@martinkennelly @kyrtapz and I were just discussing this. We came to the conclusion that this requirement shouldn't be in place. Basically we allow optional default gateways for both ipv4 and ipv6. All it means to us is whether or not we configure a default route on the OVN GR for this node. What we really want to check here is if the 2 interfaces have different interfaces. 

Furthermore, I'm not sure this is even possible to get two different interfaces. In the getDefaultGatewayInterfaceByFamily I remember I added a scope lookup to make sure we only get the routes on the interface itself:

```
			klog.Infof("Found default gateway interface %s %s", foundIfName, r.Gw.String())
			if len(gwIface) > 0 && gwIface != foundIfName {
				// this should not happen, but if it did, indicates something broken with our use of the netlink lib
				return "", nil, fmt.Errorf("mistmaching provided gw interface: %s and gateway found: %s",
					gwIface, foundIfName)
			}
```